### ISSUE TITLE
Add ContentDecoderAgent to web client howto

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5415]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5415|
|Type|enhancement|
|Created|2011-11-27 15:31:51Z|
|Branch|https://github.com/twisted/twisted/tree/contentdecoderagent-docs-5415|

`ContentDecoderAgent` is another useful `Agent`-wrapper, coming with support for one encoding (gzip).  Use of this feature should be explained and demonstrated in the web client howto.

<details><summary>Searchable metadata</summary>

```
trac-id__5415 5415
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_contentdecoderagent_docs_5415 branches/contentdecoderagent-docs-5415
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation documentation
time__1322407911000000 1322407911000000
changetime__1324301851000000 1324301851000000
version__None None
owner__exarkun exarkun
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [33384]) Branching to 'contentdecoderagent-docs-5415'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/contentdecoderagent-docs-5415)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

 - Output of the example:

```
$ python gzipdecoder.py 
Some data received:
<!-- w11.fp.ch1.yahoo.com uncompressed/chunked Mon Dec 19 00:56:35 PST 2011 -->

Finished receiving body:
```

I thought something was missing behind "Finished receiving body:" but after looking at the code it only shows an `errorMessage()` if there is one. Shouldn't this errorMessage be checked if it's empty or not and print something based on that?

 - There's a typo in the first sentence of "Automatic Content Encoding Negotiation"; requst should be request.

This can be merged after you addressed these things.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [33394]) Fix typo

refs [#5415](https://github.com/twisted/twisted/issues/5415)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

> Output of the example:

Hmm.  That's not what I get.  I suppose http://www.yahoo.com/ varies depending on... where you are?  That's a bit annoying.

> I thought something was missing behind "Finished receiving body:"

Hmm.  Yea.  It turns out the error for www.yahoo.com is `PotentialDataLoss`, though, because content-length is unset and chunked encoding is not used.  Very nice HTTP/1.1 server they've got there.  I changed it to print the exception type and value, which should at least be confusing in a different way.

> There's a typo in the first sentence of

Fixed.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [33396]) Merge contentdecoderagent-docs-5415 

Author: exarkun
Reviewer: thijs
Fixes: [#5415](https://github.com/twisted/twisted/issues/5415)

Add howto documentation and an example for `ContentDecoderAgent`.
